add keyboard search feature

iamrahulrnair opened the pull request:
Added Keyboard search functionality,User can search for a keyword and from the suggestion list user can fill the input with up and down arrow keys.

iamrahulrnair commented:
Hi can i get a review on this please :-^

Sparsh1212 commented on src/components/AdvancedSearch.js:
```diff
@@ -42,14 +45,15 @@ const AdvancedSearch = forwardRef((props, ref) => {
   const [autoComplete, setAutoComplete] = useState(null)
   const [suggestions, setSuggestions] = useState([])
   const [isInputInFocus, setIsInputInFocus] = useState(false)
+  const [new_array,setNew_array]=useState([])
+  let [counter,setCounter]=useState(0)
```
Can we give more meaningful names to these states like: `keyCounter` and something more apt for `new_array`

Sparsh1212 reviewed (changes requested):
Just some nit fixes are needed in naming the variables. Also, I need to take a proper look at your added comment. Rest everything seems fine to me.
Can you also add a demo video as a proof of working?

iamrahulrnair commented:
Hi, I have renamed the variables, and my prettier have added some auto formatting, that wont have impact on the application.  changed single quote to double quote that is all,

https://user-images.githubusercontent.com/55950758/146120882-e41a36ea-322e-417f-8119-accb4e767e0c.mp4

l

Sparsh1212 reviewed (changes requested):
Although it won't have any effect on the application still we want the codebase to be consistent throughout. Please make sure you revert all the formatting changes from this PR. You can take your time, no rush.

iamrahulrnair commented:
changed the quotes, and think the semi-colon at the end is no problem

Sparsh1212 reviewed (changes requested):
Some of the formatting changes are still not reverted like semicolons and commas. Please make sure those are reverted too.
